core/list.h

```diff
@@ -475,17 +471,13 @@ class List {
 
 		const Element *I = front();
 		int c = 0;
-		while (I) {
-
-			if (c == p_index) {
-
-				return I->get();
-			}
+		while (c < p_index) {
 			I = I->next();
 			c++;
+			CRASH_COND_MSG(!I, "Bug in list code: List shorter than size.");
```

Comment:
I wonder if we need this check. We already check that the index is within size at the start, so if we're going out of bounds here, that would be a nasty bug indeed. But maybe we can just let a potential null-pointer dereference trigger the crash instead of a manual check?

Or alternatively enclose this check in `#ifdef DEBUG_ENABLED` -- my main concern being whether this check would cost a bit when indexing a big List.

Reply:
I think it's safe to remove it. The equivalent check has been there since #8943; so I think we can assume `List` is solid.